srai/datasets/_base.py

```diff
@@ -0,0 +1,218 @@
+"""Base classes for Datasets."""
+
+import abc
+from typing import Optional
+
+import geopandas as gpd
+import h3
+import numpy as np
+import pandas as pd
+from datasets import load_dataset
+from shapely.geometry import Polygon
+from sklearn.model_selection import train_test_split
+
+from srai.regionalizers import H3Regionalizer
+
+
+class HuggingFaceDataset(abc.ABC):
+    """Abstract class for HuggingFace datasets."""
+
+    def __init__(
+        self,
+        path: str,
+        version: Optional[str] = None,
+        type: Optional[str] = None,
+        numerical_columns: Optional[list[str]] = None,
+        categorical_columns: Optional[list[str]] = None,
+        target: Optional[str] = None,
+    ) -> None:
+        self.path = path
+        self.version = version
+        self.numerical_columns = numerical_columns
+        self.categorical_columns = categorical_columns
+        self.target = target
+        self.type = type
+
+    @abc.abstractmethod
+    def _preprocessing(self, data: pd.DataFrame, version: Optional[str] = None) -> gpd.GeoDataFrame:
+        """
+        Preprocess the dataset from HuggingFace.
+
+        Args:
+            data (pd.DataFrame): a dataset to preprocess
+            version (str, optional): version of dataset
+
+        Returns:
+            gpd.GeoDataFrame: preprocessed data.
+        """
+        raise NotImplementedError
+
+    def load(
+        self, hf_token: Optional[str] = None, version: Optional[str] = None
+    ) -> gpd.GeoDataFrame:
+        """
+        Method to load dataset.
+
+        Args:
+            hf_token (str, optional): If needed, a User Access Token needed to authenticate to
+                the Hugging Face Hub. Environment variable `HF_TOKEN` can be also used.
+                Defaults to None.
+            version (str, optional): version of a dataset
+
+        Returns:
+            gpd.GeoDataFrame: Loaded data.
+        """
+        dataset_name = self.path
+        version = version or self.version
+        data = load_dataset(dataset_name, version, token=hf_token, trust_remote_code=True)
+        processed_data = self._preprocessing(data)
+
+        return processed_data
```

Comment:
I would add proper index name compatible with loaders in terms of using it later with srai